pinecone_rag/ingestion.py

```
@@ -223,106 +376,247 @@ def _is_mostly_punctuation(self, text: str) -> bool:
+                self._upsert_batch(filtered_batch, namespace, batch_num)
+                total_upserted += len(filtered_batch)
                 logger.info(
-                    f"Upserted batch {batch_num}: {len(records)}/{len(batch)} documents"
+                    "Upserted batch %s/%s: %s documents (skipped existing: %s)",
+                    batch_num,
+                    len(batch_start_idxs),
+                    len(filtered_batch),
+                    skipped_existing,
                 )
+
             except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)